soweego/linker/feature_extraction.py

```diff
@@ -327,3 +328,123 @@ def intersection_percentage_size(pair):
             return count_intersect / count_total if count_total > 0 else np.nan
 
         return fillna(concatenated.apply(intersection_percentage_size), self.missing_value)
+
+
+class OccupationCompare(BaseCompareFeature):
+
+    name = "OccupationCompare"
+    description = "Compares occupations attribute of record pairs."
+
+    # when expanding the occupations in `_expand_occupations` it
+    # is useful to have a dict were we can cache each result, so that
+    # we don't have to do a sparql query every time.
+    _expand_occupations_cache = {}
+
+    def __init__(self,
+                 left_on,
+                 right_on,
+                 missing_value=0.0,
+                 label=None):
+        super(OccupationCompare, self).__init__(left_on, right_on, label=label)
+
+        self.missing_value = missing_value
+
+    def _expand_occupations(self, occupation_qids: Set[str]) -> Set[str]:
+        """
+        This should be applied to a pandas.Series, where each element
+        is a set of QIDs.
+
+        This function will expand the set to include all superclasses and
+        subclasses of each QID in the original set.
+        """
+
+        
+        expanded_set = set()
+
+        for qid in occupation_qids:
+            expanded_set.add(qid)  # add qid to expanded set
+
+            # check if we have the subclasses and superclasses
+            # of this specific qid in memory
+            if qid in self._expand_occupations_cache:
+
+                # if we do then add them to expanded set
+                expanded_set = expanded_set | self._expand_occupations_cache[qid]
+
+            # if we don't have them then we get them from
+            # wikidata and add them to the cache and
+            # `expanded_set`
+            else:
+
+                # get subclasses and superclasses
+                subclasses = sparql_queries.get_subclasses_of_qid(qid)
+                superclasses = sparql_queries.get_superclasses_of_qid(qid)
+
+                joined = subclasses | superclasses
+
+                # add joined to cache
+                self._expand_occupations_cache[qid] = joined
+
+                # finally add them to expanded set
+                expanded_set = expanded_set | joined
+
+        return expanded_set
+
+    def _compute_vectorized(self, source_column: pd.Series, target_column: pd.Series):
+
+        # each item in `source_column` is already an array composed
+        # of QID strings representing the occupations of a person.
+        # On the other hand, each item in `target_column` is a
+        # string of space separated QIDs, so we need to first convert
+        # those into lists
+        # then we also convert each item both the source and target
+        # from arrays to sets, so that they're easier to compare
+        
+        def to_set(itm):
+            # if it is an empty array (from source), or an
+            # empty string (from target)
+            if len(itm) == 0:
```

Comment:
```suggestion
            if not itm:
```
Checking the truthy value should be enough, what do you think?